pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -253,14 +260,14 @@ func calculateIPOffset(base *big.Int, ip net.IP) int {
 	return int(big.NewInt(0).Sub(bigForIP(ip), base).Int64())
 }
 
-// RangeSize returns the size of a range in valid addresses.
-func RangeSize(subnet *net.IPNet) int64 {
+// RangeSize returns the size of a range in valid addresses or error if the mask is too short.
+func RangeSize(subnet *net.IPNet) (int64, error) {
 	ones, bits := subnet.Mask.Size()
 	if (bits - ones) >= 31 {
-		panic("masks greater than 31 bits are not supported")
+		return 0, fmt.Errorf("Subnet mask for %q is shorter than /2 for IPv4 or /98 for IPv6.", subnet)
```

Comment:
`"subnet mask must be >= /2 for IPv4 or /98 for IPv6"` (change message to be about valid values)